README.md

```diff
@@ -124,6 +124,7 @@ This will source a file `lib/file.sh` under the package `username/repo`.
 To change the behavior of basher, you can set the following variables either
 globally or before each command:
 
+- If `$XDG_DATA_HOME` is set, and `$XDG_DATA_HOME/basher` is a directory, then `$BASHER_ROOT` will be set to `$XDG_DATA_HOME/basher` instead of the usual `$HOME/.basher`. Usually, `$XDG_DATA_HOME` is set to `~/.local/share/basher`.
```

Comment:
I think there's some contradiction here. `$XDG_DATA_HOME` can't usually be `~/.local/share/basher`. Do you mean `~/.local/share/` ? Or did you mean to say that the final directory will be `~/.local/share/basher` ?

Reply:
Oops, nice catch! I also edited the description to be a bit more accurate, since we aren't actually setting `$XDG_DATA_HOME` at all

Reply:
In case `XDG_DATA_HOME` is not set, we don't do anything with it right? So it's not relevant that it should default to `~/.local/share`, since we'll not use that directory anyway by default.

Reply:
Right now, if `XDG_DATA_HOME` is not set (or has zero length), then we don't do anything with it - we instead use the value `$HOME/.local/share` instead. You're right that falling back to `$HOME/.local/share` is not relevant for people that expect existing `BASHER_ROOT` behavior (using `~/.basher`), but it still is relevant for people that are expecting XDG Base Dir behavior. If `XDG_DATA_HOME` is unset, and if we don't default to `~/.local/share`, then it will look for the Basher directory at `/basher` instead of `~/.local/share/basher`. This would be confusing because the application would look in `/basher`, whereas the user would expect the application to look in `~/.local/share/basher`

Reply:
Isn't this breaking the installation instructions in the README? It means if I follow them, the clone will be in `~/.basher` but since I didn't specify any xdg configurations, basher itself would look in the wrong place.

Reply:
Hmm, I'm not sure I follow. Like you said, after following the README instructions, the clone should be located in `~/.basher`. When Basher runs, it will first do `BASHER_ROOT="$HOME/.basher"`, which is correct. And, it will only reassign `BASHER_ROOT` to `"${XDG_DATA_HOME:-$HOME/.local/share}/basher"` if a directory exists there. Since the `README.md` doesn't say to create that directory, that codepath shouldn't be activated, so `BASHER_ROOT` should still have a value of `~/.basher`

Reply:
I did update the README explanation, since it did seem a bit confusing. Since I don't really follow you, is there anything specifically you would recommend changing (code-wise or documentation-wise)

Reply:
Ahh, I see what you mean now! This is great. Thanks for taking the time to do this!